rosbag2_performance/rosbag2_storage_plugin_comparison/scripts/sweep.py

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env python3
+""" Runs a parametric sweep of runs measuring the throughput and memory usage of
+rosbag2_storage::BaseWriteInterface implementations. Writes a CSV summary of each run to
+stdout, or to a file if a path is provided.
+"""
+
+from pathlib import Path
+import subprocess
+import shutil
+from tempfile import mkdtemp
+import csv
+import sys
+from io import StringIO
+
+import yaml
+
+"""CONFIG_DIMENSIONS contains the parameters to sweep across and their values.
+CONFIG_DIMENSIONS is structured as `{ parameter label : { variant label : config fragment } }`.
+the function `build_configs()` iterates through the variants of each parameter and produces
+a config for each combination."""
+CONFIG_DIMENSIONS = {
+    # The size and distribution of messages to write.
+    "messages": {
+        "large": { "topics": [{"name": "/large", "message_size": 1_000_000}]},
```

Comment:
I would recommend to increase default size for `"large"` messages. 
Regular use case is FulHD camera stream 1920x1080x3=622080 bytes